### ISSUE TITLE
Project questions

### ISSUE THREAD

**Issue by wrobbinz** [mattboutet/user-pal#4]
Hi @mattboutet, cool project. I'm looking into it and was hoping you could answer a couple questions for me.

1.  Why does the `/server/manifest.js` *(line 49)* reference sqlite3? My understanding is this implementation is defaulted to using postgres (which I plan to use too). Could/should this be removed?
2. Why are you using the "secure-password" module? My understanding from looking at this, is that it is a hashing library in the same vein as bcrypt. From a github-star-check/npm-download-rate (not absolute metrics, I know) it's not very popular, so I was just curious why you've chosen it. I'm hesitant to rely on it given I don't know much about it.
3. The knex module is found in the package.json's devDependencies. Shouldn't Knex be a dependency?

**Comment by mattboutet** [comment]
Hey @wrobbinz - 
Good questions, all.  
1) The sqlite3 reference is mostly vestigial code at this point.  Originally, this project was set up to be db-agnostic, but I've since moved it definitively towards postgres, so at this point using sqlite would require enough code changes that it doesn't really make sense to include that reference anymore.  I'll plan on cleaning this up soon.
2) I settled on secure-password not long after Argon2 had won the Password Hashing Competition.  I think bcrypt is also more that adequate, so it's really down to personal preference.
3) I suspect that knex won't be the only thing that comes up missing if you use `npm install --production` - that's not a normal part of my workflow, so I haven't given it much thought/testing.
